### PR TITLE
[Data] Add `SERVICE_UNAVAILABLE` to list of retried transient errors

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

While reading or writing files with Ray Data, S3 might raise a transient `SERVICE_UNAVAILABLE` error. This PR adds the error to the list of retried transient errors.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/context.py [hunk 1]
```diff
@@ -114,6 +116,7 @@
     "AWS Error NETWORK_CONNECTION",
     "AWS Error SLOW_DOWN",
     "AWS Error UNKNOWN (HTTP status 503)",
+    "AWS Error SERVICE_UNAVAILABLE",
```
Comment (scottjlee) [inline python/ray/data/context.py:119]: should we also add this to `DEFAULT_RETRIED_IO_ERRORS`?

**Review by scottjlee — APPROVED** [review]

File: python/ray/data/context.py (hunk 1 above)
Comment (bveeramani) [reply python/ray/data/context.py:119]: It is in it. You might need to expand the diff
image

**Review by omatthew98 — APPROVED** [review]

File: python/ray/data/context.py (hunk 1 above)
Comment (scottjlee) [reply python/ray/data/context.py:119]: oh sorry i got confused by the top comment above `DEFAULT_WRITE_FILE_RETRY_ON_ERRORS`. for backwards compatibility, should we also add it in `DEFAULT_WRITE_FILE_RETRY_ON_ERRORS`?

Comment (bveeramani) [reply python/ray/data/context.py:119]: `DEFAULT_WRITE_FILE_RETRY_ON_ERRORS` isn't used anywhere right now. If you try to configure `write_file_retry_on_errors`, Ray Data emits a warning instructed you to configure `retried_io_errors` instead.

**Review by scottjlee — APPROVED** [review]

**Pull request merged**